OgreMain/include/Ogre.i

```diff
@@ -547,6 +575,29 @@ SHARED_PTR(Material);
     %template(SubEntityList) std::vector<Ogre::SubEntity*>;
     %ignore Ogre::Entity::getAttachedObjectIterator;
     %include "OgreEntity.h"
+	%inline %{
+		static void* SWIG_CastMovableObjectToEntity(void * jarg1)
+		{
+          void * jresult ;
+          Ogre::MovableObject *arg1 = (Ogre::MovableObject *) 0 ;
+          Ogre::Entity *result = 0 ;
+
+          arg1 = (Ogre::MovableObject *)jarg1;
+          {
+            try {
+              result = (Ogre::Entity *)arg1;
```

Comment:
this will never throw. You probably wanted to do that:
```cpp
result = dynamic_cast<Ogre::Entity*>(arg1);
if(!result)
{
    SWIG_CSharpException ...
}
```